src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -188,3 +189,35 @@ get_outgoing_pools() ->
     [#{report_name => outgoing_pools,
        key => type,
        value => Type} || {Type, _, _, _, _} <- OutgoingPools].
+
+get_xmpp_stanzas_count() ->
+    Hosts = ejabberd_config:get_global_option(hosts),
+    StanzaTypes = [xmppMessageSent, xmppMessageReceived, xmppIqSent,
+                   xmppIqReceived, xmppPresenceSent, xmppPresenceReceived],
+    NewCount = [count_stanzas(Hosts, StanzaType) || StanzaType <- StanzaTypes],
+    PrevReport = mongoose_system_metrics_file:read(),
```

Comment:
How hard would it be to store the prev value in the state of the main process triggering the metrics collection? I'm not sure if read from file is the best solution, the file can be removed or changed at any time.

Reply:
Returning just these values wouldn't be that hard, just could be a bit messy. What do you think of storing full last report in the state of the main process? Collecting metrics would take Previous report as an argument.

Reply:
I think we could store in the state only these values which are needed for the next report. I'm not sure there is a need to store full report. On the other hand storing full report may indeed be cleaner.

Reply:
Storing full report would make it easier to understand on what is stored there. It could either be 
* full report
* short report with values that might be needed for the next one
* explicit tuples {metric, count} and having each value stored explicitly in the state
which do you think would make most sense to implement?

Reply:
Let's go for the simplicity and store the full report in the state.